File: frontend/src/services/contactRequestService.js
import api from './api'

export const contactRequestService = {
  async createContactRequest(payload) {
    return await api.post('/contact-requests', payload)
  },

  async getContactRequests() {
    return await api.get('/contact-requests')
  },

  async getContactRequest(id) {
    return await api.get(`/contact-requests/${id}`)
  },

  async deleteContactRequest(id) {
    return await api.delete(`/contact-requests/${id}`)
  },
}

File: backend/internal/domain/services/contact_request_service.go
package services

import "ccpp-backend/internal/domain/models"

type ContactRequestService interface {
	CreateContactRequest(request *models.ContactRequest) error
	GetContactRequests() ([]models.ContactRequest, error)
	GetContactRequest(id uint) (*models.ContactRequest, error)
	DeleteContactRequest(id uint) error
}
